drivers/iio/adc/ad4630.c

```diff
@@ -590,26 +598,32 @@ static int ad4630_write_raw(struct iio_dev *indio_dev,
 	}
 }
 
-static int ad4630_update_sample_fetch_trigger(const struct ad4630_state *st, u32 avg)
+static int ad4630_update_sample_fetch_trigger(struct ad4630_state *st, u32 avg)
 {
-	struct pwm_state fetch_state, conv_state;
+	struct pwm_waveform fetch_wf = { };
```

Comment:
```suggestion
	struct pwm_waveform fetch_wf = st->fetch_wf;
```

Then we don't need to add `fetch_wf.duty_length_ns = st->fetch_wf.duty_length_ns;`.